File: scripts/delete_entries.py
# scripts/delete_entries.py
from pathlib import Path
import argparse
from file_handlers import StringsFileHandler, JsonFileHandler

class StringsEditor:
    def __init__(self):
        self.strings_handler = StringsFileHandler()
        self.json_handler = JsonFileHandler()
        self.base_path = Path("./Sources/CasaZurigol10n/Resources")

    def delete_entry(self, file_path, key_to_delete):
        """Delete entry from a file (.strings or .json)"""
        path = Path(file_path)
        handler = self.strings_handler if path.suffix == '.strings' else self.json_handler
        
        if path.exists() and path.is_file():
            translations = handler.parse_file(path)
            if key_to_delete.lower() in translations:
                del translations[key_to_delete.lower()]
                handler.create_file(translations, path)
                print(f"Deleted key '{key_to_delete}' from {path}")
                return True
        return False

    def delete_entry_from_all_files(self, key_to_delete):
        """Delete entry from all .strings and .json files in the Resources directory"""
        deletions_count = 0
        
        # Find and process all .strings and .json files
        for file_path in self.base_path.rglob("*"):
            if file_path.suffix in ['.strings', '.json']:
                if self.delete_entry(file_path, key_to_delete):
                    deletions_count += 1
        
        return deletions_count

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Delete entries from localization files")
    parser.add_argument(
        "--key",
        required=True,
        help="The key to delete from all files"
    )

    args = parser.parse_args()
    editor = StringsEditor()
    deletions = editor.delete_entry_from_all_files(args.key)
    print(f"\nTotal deletions performed: {deletions}")

File: scripts/deleteEntries.ts
import { Command } from "commander";
import path from "path";
import fs from "fs";
import { FileHandlerFactory } from "./fileHandlers";

class StringsEditor {
  private basePath: string;

  constructor(basePath: string) {
    this.basePath = basePath;
  }

  deleteEntry(filePath: string, keyToDelete: string): boolean {
    try {
      const handler = FileHandlerFactory.getHandler(filePath);
      if (fs.existsSync(filePath) && fs.statSync(filePath).isFile()) {
        const translations = handler.parseFile(filePath);
        const lowerKey = keyToDelete.toLowerCase();

        if (lowerKey in translations) {
          delete translations[lowerKey];
          handler.createFile(translations, filePath);
          console.log(`Deleted key '${keyToDelete}' from ${filePath}`);
          return true;
        }
      }
      return false;
    } catch (err) {
      console.error(`Error deleting key in file '${filePath}': ${err}`);
      return false;
    }
  }

  deleteEntryFromAllFiles(keyToDelete: string): number {
    let deletionsCount = 0;

    const processDirectory = (dirPath: string) => {
      const files = fs.readdirSync(dirPath);

      files.forEach((file) => {
        const fullPath = path.join(dirPath, file);
        const stat = fs.statSync(fullPath);

        if (stat.isDirectory()) {
          processDirectory(fullPath);
        } else {
          if (this.deleteEntry(fullPath, keyToDelete)) {
            deletionsCount++;
          }
        }
      });
    };

    processDirectory(this.basePath);
    return deletionsCount;
  }
}

const program = new Command();

program
  .requiredOption("--key <key>", "The key to delete from all files")
  .parse(process.argv);

const basePath = "./Sources/CasaZurigol10n/Resources";
const options = program.opts();
const editor = new StringsEditor(basePath);
const deletions = editor.deleteEntryFromAllFiles(options.key);
console.log(`\nTotal deletions performed: ${deletions}`);
